pkg/ccl/changefeedccl/cdctest/validator.go

```diff
@@ -222,30 +223,39 @@ func (v *beforeAfterValidator) NoteRow(
 			`expected primary key columns %s got datums %s`, v.primaryKeyCols, primaryKeyDatums)
 	}
 
-	type wrapper struct {
-		After  map[string]interface{} `json:"after"`
-		Before map[string]interface{} `json:"before"`
+	j, _ := json.ParseJSON(value)
+	afterJson, err := j.FetchValKey("after")
+	if err != nil {
+		return err
+	}
+	afterDatums, err := parseJSON(afterJson)
```

Comment:
nit: naming: these arent datums, they're "go values"

Reply:
Renamed it to convertJSONToMap. Do you like this more?

Reply:
Discussed this offline. The name for Datums was adopted because we pass it as a `RowDatum` https://github.com/cockroachdb/cockroach/blob/587066079df5fac4781acb4b0ead792ad561c215/pkg/ccl/changefeedccl/cdctest/validator.go#L262 here. So we are cool with leaving Datums. I renamed this to `afterValueDatums` to be a bit clearer.